nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/listener/ElasticSearchInlineListener.java

```diff
@@ -106,10 +106,12 @@ public void beforeCompletion() {
 
     @Override
     public void afterCompletion(int status) {
-        if (getAllCommands().isEmpty()) {
-            return;
-        }
+        // un hook the current listener even if there's no commands to index
+        // unless, during next transaction this listener won't be hooked to it
```

Comment:
Now we update document root during cleanupSession. Which triggers an UPDATE event, if we don't index the root document, there'll be no commands to index but the listener is still hooked to the current transaction and do not un-hook. During next test, we still use the same thread, listener will have `isEnlisted.get() == true` and it won't be hooked to current transaction.